core/services/gateway/connector/connector.go

```diff
@@ -71,24 +72,27 @@ func (c *gatewayConnector) Name() string { return c.lggr.Name() }
 
 type gatewayState struct {
 	conn     network.WSConnectionWrapper
-	config   ConnectorGatewayConfig
+	config   config.ConnectorGatewayConfig
 	url      *url.URL
 	wsClient network.WebSocketClient
 }
 
-func NewGatewayConnector(config *ConnectorConfig, signer Signer, handler GatewayConnectorHandler, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
-	if config == nil || signer == nil || handler == nil || clock == nil || lggr == nil {
+func NewGatewayConnector(configLocal *config.ConnectorConfig, signer Signer, handler GatewayConnectorHandler, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
```

Comment:
@DavidOrchard Your problem is that you're passing in a pointer to an interface and then expecting to use the interface itself.

Interfaces are a bit strange in Golang: they represent a container type over a concrete type and for that reason are typically always passed around by value. The container itself could contain a pointer type though.

You should be able to fix this if you replace -> *config.ConnectorConfig with config.ConnectorConfig

Reply:
I see the original (connector's)ConnectorConfig is a struct.  If I want to preserve using a struct, does that mean I should use toml.ConnectorConfig instead of config.ConnectorConfig?  Is that appropriate?  cc @justinkaseman

Reply:
From @cedric-cordenier via slack `For config we usually pass around interfaces
t is a bit odd as a pattern so your instincts are right, but it's correct in this context`